modules/appeals_api/app/models/appeals_api/supplemental_claim.rb

```diff
@@ -31,6 +31,120 @@ def self.date_from_string(string)
       if: proc { |a| a.form_data.present? }
     )
 
+    def pdf_structure(version)
+      Object.const_get(
+        "AppealsApi::PdfConstruction::SupplementalClaim::#{version.upcase}::Structure"
+      ).new(self)
+    end
+
+    def veteran_first_name
+      auth_headers.dig('X-VA-First-Name')
+    end
+
+    def veteran_middle_initial
+      auth_headers.dig('X-VA-Middle-Initial')
+    end
+
+    def veteran_last_name
+      auth_headers.dig('X-VA-Last-Name')
+    end
+
+    def full_name
+      "#{veteran_first_name} #{veteran_middle_initial} #{veteran_last_name}".squeeze(' ').strip
+    end
+
+    def ssn
+      auth_headers.dig('X-VA-SSN')
+    end
+
+    def file_number
+      auth_headers.dig('X-VA-File-Number')
+    end
+
+    def veteran_dob_month
+      birth_date.strftime '%m'
+    end
+
+    def veteran_dob_day
+      birth_date.strftime '%d'
+    end
+
+    def veteran_dob_year
+      birth_date.strftime '%Y'
+    end
+
+    def veteran_service_number
+      auth_headers.dig('X-VA-Service-Number')
+    end
+
+    def insurance_policy_number
+      auth_headers.dig('X-VA-Insurance-Policy-Number')
+    end
+
+    def mailing_address_number_and_street
+      veteran.dig('address', 'addressLine1') || ''
+    end
+
+    def mailing_address_apartment_or_unit_number
+      veteran.dig('address', 'addressLine2') || ''
+    end
+
+    def mailing_address_box
+      veteran.dig('address', 'addressLine3') || ''
+    end
+
+    def mailing_address_city
+      veteran.dig('address', 'city') || ''
+    end
+
+    def mailing_address_city_and_box
+      "#{mailing_address_city} #{mailing_address_box}"
+    end
+
+    def mailing_address_state
+      veteran.dig('address', 'stateCode') || ''
+    end
+
+    def mailing_address_country
+      veteran.dig('address', 'countryCodeISO2') || ''
+    end
+
+    def zip_code
+      if zip_code_5 == '00000'
+        veteran.dig('address', 'internationalPostalCode') || '00000'
+      else
+        zip_code_5
+      end
+    end
+
+    def zip_code_5
+      veteran.dig('address', 'zipCode5') || '00000'
+    end
+
+    def phone
+      veteran_phone.to_s
+    end
+
+    def veteran_phone_data
+      veteran&.dig('phone')
+    end
+
+    def email
+      veteran&.dig('emailAddressText').to_s.strip
+    end
+
+    def consumer_name
+      auth_headers&.dig('X-Consumer-Username')
+    end
+
+    def consumer_id
+      auth_headers&.dig('X-Consumer-ID')
+    end
+
+    def benefit_type
+      data_attributes&.dig('benefitType').to_s.strip
```

Comment:
is benefitType likely to not be a string? Was there a use case here for `to_s`?

Reply:
Good catch. No, I can't see why coercing a string is necessary here. Shall fix!